virtualmachine/vsphere/vm.go

```diff
@@ -726,3 +726,38 @@ func (vm *VM) GetSSH(options ssh.Options) (ssh.Client, error) {
 	client := ssh.SSHClient{Creds: &vm.Credentials, IP: ips[0], Port: 22, Options: options}
 	return &client, nil
 }
+
+func DeleteTemplate(vm *VM) error {
+	templates := make([]string, 0)
+	if err := SetupSession(vm); err != nil {
+		return err
+	}
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
+	}
+	for _, datastore := range vm.Datastores {
+		template := createTemplateName(vm.Template, datastore)
+		temp, err := findVM(vm, dcMo, template)
+		if err != nil {
+			templates = append(templates, template)
```

Comment:
Based on the value of flag UseLocalTemplates in vm struct the template may be uplaoded to all the datastores passed in. 

	// UseLocalTemplates is a flag to indicate whether a template should be uploaded on all
	// the datastores that were passed in.